### PR TITLE
Some very minor refactorings, improvements and compiler warnings fixes

### PR DESCRIPTION
**Pull request by bookmist**
Not sure in "app/screen_render: add stensil clear before draw"
and "TLS init fix" but tls init fix is better then before (but may be still buggy)

### CODE REVIEW

File: vita3k/kernel/src/thread/thread.cpp [hunk 1]
```diff
@@ -116,11 +116,11 @@ SceUID create_thread(Ptr<const void> entry_point, KernelState &kernel, MemState
 
     alloc_name = fmt::format("TLS for thread {} (#{})", name, thid);
 
-    auto tls_address = alloc(mem, 0x800, alloc_name.c_str()) + 0x800;
+    Ptr<void> tls_address_ptr = Ptr<void>(alloc(mem, 0x800, alloc_name.c_str()));
```
Comment (Zangetsu38) [inline vita3k/kernel/src/thread/thread.cpp:119]: you can also using auto here, you using Ptr for intial value so is good on auto too :)

**Comment by Zangetsu38** [comment]
no take your head, jun run format :D and set format change inside commit relate :)

**Comment by sunho** [comment]
Can you tell me little more about tls init fix?

File: vita3k/app/src/screen_render.cpp [hunk 2]
```diff
@@ -102,7 +102,7 @@ void gl_screen_renderer::render(const HostState &host) {
     glViewport(static_cast<GLint>(host.viewport_pos.x), static_cast<GLint>(host.viewport_pos.y), static_cast<GLsizei>(host.viewport_size.x),
         static_cast<GLsizei>(host.viewport_size.y));
 
-    glClear(GL_COLOR_BUFFER_BIT | GL_DEPTH_BUFFER_BIT);
+    glClear(GL_COLOR_BUFFER_BIT | GL_DEPTH_BUFFER_BIT | GL_STENCIL_BUFFER_BIT);
```
Comment (pent0) [inline vita3k/app/src/screen_render.cpp:105]: We don't use stencil in main framebuffer, so it's no need. What is the reason?

Comment (bookmist) [reply vita3k/app/src/screen_render.cpp:105]: I had yellow and blue screen on loading ar nosurge game. At some time this change and maybe some more half-random changes replace yellow to black (that, I think, is more like original behavior). Now i rechecked this and see that this change only has no significant effect.
Thank you very much for explaining, I'll revert this commit.

**Comment by bookmist** [comment]
> Can you tell me little more about tls init fix?

About tls_fix. Before my commit emu did the following:
alloc 0x800 bytes of memory
set pointer to the end of allocated memory
memset from this pointer (END of allocated mem) 0x800 bytes to zero

I changed this to memset from BEGIN of allocated mem.

I'm not sure should I init this memory with kernel.tls_address* and should I allocate 0x800 bytes or kernel.tls_msize bytes.
Also maybe function get_thread_tls_addr should be simplified and kernel.tls field can be deleted.

**Comment by sunho** [comment]
You are totally right. Nice find!

If kernel.tls_address is not empty, we should memcopy from the BEGINNING of allocated memory. And, we should allocate 4* kernel.tls_msize in stead of 0x800 in this case. 

I agree that we can just simplify get_tls_address. kernel.tls field is merely serving as cache. No need for caching this kind of thing.

**Comment by sunho** [comment]
As I didn't write that code on tls, I'll ask frangarj who wrote that code just in case to confirm. But, it's fairly clear we should do that from beginning of that area looking at old code.

**Review by sunho — APPROVED** [review]
We should merge this soon. Faulty tls is a huge deal.

**Comment by sunho** [comment]
@bookmist tell me if I can merge this.

**Comment by bookmist** [comment]
@sunho Yes, of course.

**Pull request merged**